### DISCUSSION TITLE
RaspberryPi alternative?

### DISCUSSION THREAD

**Discussion by qdmt** [outdoorbits/little-backup-box#189] [Q&A] [answered]
Given that the pricing and supply issues with these devices, can this solution be implemented with another similarly-performing single board PC?

**Comment by outdoorbits** [comment, marked as answer]
Did you read the great report of [rkosick](https://github.com/rkosick) in [Where have you been traveling together with Little Backup Box? ](https://github.com/outdoorbits/little-backup-box/discussions/155)? He used a Raspberry Pi Zero 2W in combination with a Zero4U 4-Port USB Hub. This looks like a great combination, roundabout at half the price of a pi 4.
It's to expect, newer functions like creating thumbnails from raw-fotos would be significant slower. There is just a discussion about transfer speed.. Maybe the disadvantage of USB 2.0 vs. 3.0 is by far not as relevant as it seems. This means, a pi zero or pi 3 might be a good alternative.
There are no reports about some alternatives like orange pi...

**Reply by qdmt** [reply]
I had read that report some time back, but completely glossed over that he was using a Zero 2W. My bad! But this is excellent - certainly expands the option base.

**Comment by outdoorbits** [comment]
Now I installed lbb on an old Raspberry Pi 2 Model B Rev 1.1.

Everything works fine, even the display.

On backup from USB to internal I got transfer-rates from 10 to 13 MB/s. 64 GB should be copied in 1:30 hours.
Converting (raw)images to thumbnails was also slow: In my try I had pairs of jpeg and raw from a Nikon Z6. The average time to create one thumbnail was about 100 seconds.

No one of the actual models of the pi should be as slow as this one.

**Reply by qdmt** [reply]
Great, this is good - thank you for testing. Now with more options I should be able to find something reasonably priced. BTW - not sure if it helps to know - but I'm planning to use a 1TB SSD OPAL 2.0 PCIe-NVMe that is enclosed in a Realtek RTL9210B-CG (ORICO dual protocol ssd enclosure). A surplus SSD I have that I think will be a great candidate for my first LBB.

**Reply by qdmt** [reply]
@outdoorbits So I got my LBB up and running (found a RPi4MB on auction at a very reasonable price). However, the 1TB SSD OPAL 2.0 PCIe-NVMe that is enclosed in a Realtek RTL9210B-CG (ORICO dual protocol ssd enclosure) is exhibiting weird behaviour.

I had it initially plugged in directly, then also tried a powered KVM switch.

It reads correctly as a target, with correct size and all, but then it will begin transferring from a source and then at some point fail and say something to the effect of "not enough voltage, CPU is throttled".

Rebooting doesn't work. I went to bed, woke up and tried again - then same thing happens - it discovers the 1TB SSD again but fails due to insufficient voltage.

How do I diagnose or address this?

**Reply by rkosick** [reply]
What kind of power supply are you using? Generic USB power supplys are often not working well with the RP4. Should not be an issue with the power consumption of your SSD since you testet it with a powered switch.
Have you tried an external power bank as a power supply for the RP4? I never had issues using any of those.

**Reply by qdmt** [reply]
> What kind of power supply are you using? Generic USB power supplys are often not working well with the RP4. Should not be an issue with the power consumption of your SSD since you testet it with a powered switch.

I've tried various - but currently I am trying a white iPad-style adapter that says 5v≈3A on it with a USB C-to-C cable.

> Have you tried an external power bank as a power supply for the RP4? I never had issues using any of those.
Not yet - none of my power banks are 3.0A (yet).

**Reply by rkosick** [reply]
It's not about just Voltage and Current writen on the adapter. Most of them just can't deliver high current for a short amount of time and that's the problem. It's because they are produced cheap. Don't worry about a Powerbank with less than 3A. Even under load you won't need that power.

**Reply by qdmt** [reply]
It's always a victory when I can finally the 1TB to read, but still can't get it to finish a process:

08:03:55
mount_device RESULT=--uuid 74A20D26A20CEF04

08:03:56
| Target o.k.
| size: 1000M
| used: 967M, 97%
| free: 34M
| FS: fuseblk

08:03:56
| Ready
| Insert source
| Target o.k.
| size: 1000M
| used: 967M, 97%

08:04:01
| IP (offline):
| 10.41.0.1
| Ready
| Insert source
| Target o.k.

08:05:09
DEVICE_CHOSEN_IDENT='--uuid 564B-8C6A' ('/dev/sdb1') prepared to mount at '/media/source', exfat

08:05:10
| mounting:
| USB-source
| IP (offline):
| 10.41.0.1
| Ready

08:05:10
mount_device RESULT=--uuid 564B-8C6A

08:05:11
| Source o.k.
| Working...
| size: 102G
| used: 200M, 1%
| FS: exfat

08:05:11
Files to sync from folder '/media/source': 6

08:05:44
| USB storage 
|  > USB storage 
| 0 of 6
| Time: ?
| 0% ____________________

08:05:45
| USB storage 
|  > USB storage 
| 0 of 6
| Time: ?
| comparing...

08:05:47
/var/www/little-backup-box/tmp/sync.log:
>>>>>

2023/02/11 08:05:43 [10740] building file list
2023/02/11 08:05:43 [10740] .d..t...... ./
2023/02/11 08:05:45 [10740] >f+++++++++ NIKON001.DSC
2023/02/11 08:05:45 [10740] cd+++++++++ DCIM/
2023/02/11 08:05:45 [10740] cd+++++++++ DCIM/100NCZ_6/
2023/02/11 08:05:47 [10740] rsync: [sender] write error: Broken pipe (32)
<<<<<

08:05:48
Files missing: 5 files not synced.

08:05:48
The voltage is currently too low.
CPU is currently throttled.

08:05:48
Exception: 11 11=Error in file I/O

08:05:48
Power has previously been to low.
CPU has previously been throttled.

08:05:49
| Try backup 2 of 5
| USB storage 
|  > USB storage 
| 0 of 6
| Time: ?

08:05:54
Files to sync from folder '/media/source': 5

08:05:55
| USB storage 
|  > USB storage 
| 0 of 5
| Time: ?
| 0% ____________________

08:06:02
| IP (offline):
| 10.41.0.1
| USB storage 
|  > USB storage 
| 0 of 5

08:06:21
| USB storage 
|  > USB storage 
| 0 of 5
| Time: ?
| comparing...

08:06:22
/var/www/little-backup-box/tmp/sync.log:
>>>>>
2023/02/11 08:05:54 [11433] building file list
2023/02/11 08:06:21 [11433] .d..t...... DCIM/100NCZ_6/
2023/02/11 08:06:22 [11433] rsync: [sender] write error: Broken pipe (32)
<<<<<

08:06:23
Files missing: 5 files not synced.

08:06:23
Power has previously been to low.
CPU has previously been throttled.

08:06:23
Exception: 11 11=Error in file I/O

08:06:23
Power has previously been to low.
CPU has previously been throttled.

08:06:24
| Try backup 3 of 5
| USB storage 
|  > USB storage 
| 0 of 5
| Time: ?

08:06:29
Files to sync from folder '/media/source': 5

08:06:30
| USB storage 
|  > USB storage 
| 0 of 5
| Time: ?
| 0% ____________________

08:07:01
| IP (offline):
| 10.41.0.1
| USB storage 
|  > USB storage 
| 0 of 5

08:07:25
| USB storage 
|  > USB storage 
| 0 of 5
| Time: ?
| comparing...

08:07:27
/var/www/little-backup-box/tmp/sync.log:
>>>>>
2023/02/11 08:06:29 [12415] building file list
2023/02/11 08:07:24 [12415] .d..t...... DCIM/100NCZ_6/
2023/02/11 08:07:27 [12415] rsync: [sender] write error: Broken pipe (32)
<<<<<

08:07:28
Files missing: 5 files not synced.

08:07:28
Power has previously been to low.
CPU has previously been throttled.

08:07:28
Exception: 11 11=Error in file I/O

08:07:28
Power has previously been to low.
CPU has previously been throttled.

08:07:29
| Try backup 4 of 5
| USB storage 
|  > USB storage 
| 0 of 5
| Time: ?

08:07:34
Files to sync from folder '/media/source': 5

08:07:35
| USB storage 
|  > USB storage 
| 0 of 5
| Time: ?
| 0% ____________________

**Reply by qdmt** [reply]
> It's not about just Voltage and Current writen on the adapter. Most of them just can't deliver high current for a short amount of time and that's the problem. It's because they are produced cheap. Don't worry about a Powerbank with less than 3A. Even under load you won't need that power.

Great - thanks - I'll give it a shot.

**Reply by qdmt** [reply]
So I was able to successfully complete a transfer. Not exactly sure about the conditions that made it work, but will keep repeating and note down the protocol.

**Reply by qdmt** [reply]
Target 1TB SSD seems to work when it's in this cable-free enclosure:
[https://a.co/d/akssiB0]

But not the ORICO USB C-to-C cabled enclosure (regardless of cable):
[https://www.amazon.ca/dp/B08688V285]

Was able to complete one successful backup, but when trying again with the same source and target (to see if it would recognize that it is synced already) it went the opposite direction (target to source) AND pulled in folders from a difference source many years ago (guess it was in internal cache somehow).

Will try to recap my tests in a more structured manner on a new thread as I think this thread's original intent is resolved.

**Comment by outdoorbits** [comment]
This again looks like a power-shortage. lbb uses the first connected device as target, the next as source. After connecting the first USB-device, linux assigns a LUM-alpha to it, let's say sda. The next device will get sdb. If sda disconnects (maybe because of low voltage), next time it will not be assigned as sda again but as sdc. Now sdb is target, sdc is source - opposite direction.
If a re-connection happens while a backup is running, lbb will catch that by the devices unique UUIDs.
To ensure the desired backup-direction, just disconnect your source device before you start the backup an connect it, when you are asked to do this.

**Reply by qdmt** [reply]
Ah! I knew there had to be a rhyme to it - apologies if I've missed that in the documentation.
